src/Microsoft.DotNet.Build.Tasks/VersionTools/BaseDependenciesTask.cs

```diff
@@ -261,6 +261,14 @@ private FileUpdater ConfigureFileUpdater(FileUpdater updater, ITaskItem step)
                 "true",
                 StringComparison.OrdinalIgnoreCase);
 
+            string oldValue = step.GetMetadata("ReplacementSubstituteOld");
+            string newValue = step.GetMetadata("ReplacementSubstituteNew");
```

Comment:
Should we log a warning/error if only one is specified and not the other?

Reply:
Yeah, adding. Thanks.

Reply:
`GetMetadata` actually returns empty string rather than null if the value isn't specified, so I'm just adding an error case for `newValue` but no `oldValue`. `oldValue` with no `newValue` means replacing something with the empty string.